### ISSUE TITLE
Proposal to refactor Config for User Interface Testing

### ISSUE THREAD

**Issue by eviltofu** [AlphaWallet/alpha-wallet-ios#3726]
Config needs to be able to 

1. Have it's own namespace for uiTesting (perhaps by adding `ui-testing` in front of the `UserDefault` keys.
2. Create several sets of fixed Wallets for uiTesting.
3. Passed from one coordinator to another instead of just creating it on the coordinator's `init` method.

**Comment by hboon** [comment]
> 3. Passed from one coordinator to another instead of just creating it on the coordinator's init method

Definitely. We have removed many of these. But appears to have come back. Maybe a PR to get rid of these first? Might involve a number of files.

**Comment by hboon** [comment]
> 1. Have it's own namespace for uiTesting (perhaps by adding ui-testing in front of the UserDefault keys.

Since `Config` mostly reads from `UserDefaults` [^1], maybe it's easier to just add a initializer `init(defaults: UserDefaults)`, change `Config` to use it and pass in a different `UserDefaults` instance for tests?

[^1]: we should probably see if we can fix any values that aren't.

**Comment by hboon** [comment]
Actually, we are already halfway there: https://github.com/AlphaWallet/alpha-wallet-ios/blob/c98072d2771fe4f3d277c2e7e9a07916470a3663/AlphaWalletTests/Factories/Config.swift#L7

**Comment by hboon** [comment]
Maybe it's the static functions in `Config` which are the problem

**Comment by hboon** [comment]
@eviltofu do you have enough feedback from me?
